### PR TITLE
Make the filesystem writeable in Terraform

### PR DESCRIPTION
**Pull request by inercia**
We are having problems installing packages with transactional-update in Terraform, as TF doesn't like the reboot. So make the filesystem read-writeable instead and instyall with a regular "zypper install". This should be gone once we have the required packages in the base OS...

### CODE REVIEW

File: deployments/tf-libvirt-nodes/terraform.tf [hunk 1]
```diff
@@ -192,19 +204,21 @@ resource "libvirt_domain" "node" {
...
+  # TODO: we must install docker, kubelet, etc... remove this once we have them in the base image
   provisioner "remote-exec" {
     inline = [
       "for repo in /etc/zypp/repos.d/*.repo ; do echo 'gpgcheck=off' >> $repo ; done",
-      "transactional-update --no-selfupdate pkg install -y kubernetes-kubelet",
-      "echo 'Rebooting with the new packages...' && shutdown -r now",
+      "btrfs property set -ts /.snapshots/1/snapshot ro false",
+      "mount -o remount,rw /",
+      "zypper ref",
+      "zypper install -y ${var.rpms}",
```
Comment (sysrich) [inline deployments/tf-libvirt-nodes/terraform.tf]: While I don't like this, if you're going to do it, please at least restore the read-only mounting so all your testing after the package install is at least somewhat reflective of real world systems.

File: deployments/tf-libvirt-nodes/terraform.tf [hunk 2]
```diff
@@ -84,6 +84,18 @@ variable "password" {
   description = "password for sshing to the VMs"
 }
 
+variable "rpms" {
+  type = "string"
+  default = "kubernetes-kubelet docker-kubic"
```
[the remark this answers is not in the document]
Comment (inercia) [reply deployments/tf-libvirt-nodes/terraform.tf]: These RPMs installation will be gone once we have a stable base image with all the things we need. And switching to a different image could be done by pointing [this variable](https://github.com/kubic-project/kubic-init/blob/eeaf338c20bab5a302e0161e8f7f8fe16d4d330e/deployments/tf-libvirt-full/terraform.tf#L30) to a different URL.

File: deployments/tf-libvirt-nodes/terraform.tf [hunk 3]
```diff
@@ -30,7 +30,7 @@ variable "img_pool" {
 variable "img_url_base" {
   type = "string"
   #default     = "https://download.opensuse.org/repositories/devel:/kubic:/images/images/"
-  default     = "https://download.opensuse.org/repositories/devel:/CaaSP:/images/images/"
+  default = "https://download.opensuse.org/repositories/devel:/CaaSP:/images/images/"
```
[the remark this answers is not in the document]
Comment (inercia) [reply deployments/tf-libvirt-nodes/terraform.tf]: Probably. We will change it once we have a stable base image.

File: deployments/tf-libvirt-nodes/terraform.tf [hunk 4]
```diff
@@ -192,19 +204,21 @@ resource "libvirt_domain" "node" {
...
   }
 
-  # TODO: we must install docker... remove this
+  # TODO: we must install docker, kubelet, etc... remove this once we have them in the base image
   provisioner "remote-exec" {
     inline = [
       "for repo in /etc/zypp/repos.d/*.repo ; do echo 'gpgcheck=off' >> $repo ; done",
-      "transactional-update --no-selfupdate pkg install -y kubernetes-kubelet",
-      "echo 'Rebooting with the new packages...' && shutdown -r now",
+      "btrfs property set -ts /.snapshots/1/snapshot ro false",
```
[the remark this answers is not in the document]
Comment (inercia) [reply deployments/tf-libvirt-nodes/terraform.tf]: AFAIK this is the "recommended" way to make the filesystem writeable (at least we have been doing this so far). I would prefer to keep it as it is...

File: deployments/tf-libvirt-nodes/terraform.tf (hunk 1 above)
Comment (inercia) [reply deployments/tf-libvirt-nodes/terraform.tf]: The filesystem cannot be restored to _read-only_ state once it has been made _writeable_.

**Comment by inercia** [comment]
> Despite we are in a weird situation with images at the moment, we have never discussed how the "development environment" should be; this is what we have at the moment because it was created along with the PoC. I think we should at some moment also talk about the development environment.

> I start to feel that we are doing many hacks just to get an environment going, yet while the goal is to depend on almost nothing on the base system (cri-o, podman). At the moment it's possible to work with the Kubic ISO (use autoyast if needed) and snapshotting should be a good strategy to recover old state if needed.

@ereslibre Please feel free to add any other deployment mechanism you consider appropriate in the `/deployments` directory. I guess this is question of  personal taste, but I've been using Terraform/libvirt with snapshots so far and it works fine for me. Having alternative deploymebt mechanism basedc on ISOs could be useful too, as users will probably deploy Kubic in many different ways.

**Comment by inercia** [comment]
I think we can conclude that this PR can be merged. There are multiple `TODO`s here that should be fixed in the short term, but in the meantime we must have some functional (although hackish) Terraform deployment. We add to the TODO list that we should include some deployment based on ISOs, so we can play with configuration comming from YaST...

**Pull request merged**